### PR TITLE
refactor(devtools): devtools support for maps and symbol description

### PR DESCRIPTION
**Pull request by ducin**
Added 2 tiny improvements:
- instead of "Symbol()", "Symbol(DESCRIPTION)" is displayed
- ECMAScript Maps are distinguished

Additionally:
- PropTypes has been moved to a separate file
- Simple unit tests covering each PropType except for PropType.Unknown

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

- ECMAScript Maps are not distinguished in devtools
- Symbols are displayed, but their description is not

Issue Number: N/A

## What is the new behavior?

- ECMAScript Maps are displayed, similar as Sets
- Symbols are displayed along with their description

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by ducin** [comment]
@JeanMeche will appreciate if you could take a look

File: devtools/projects/ng-devtools-backend/src/lib/state-serializer/serialized-descriptor-factory.ts [hunk 1]
```diff
@@ -54,23 +55,24 @@ const serializable: {[key in PropType]: boolean} = {
...
-  [PropType.Date]: (prop: any) => {
+  [PropType.Object]: (prop: Object) => (getKeys(prop).length > 0 ? '{...}' : '{}'),
+  [PropType.Symbol]: (symbol: symbol) => `Symbol(${symbol.description})`,
+  [PropType.Undefined]: (_: undefined) => 'undefined',
+  [PropType.Date]: (prop: unknown) => {
     if (prop instanceof Date) {
       return `Date(${new Date(prop).toISOString()})`;
     }
-    return prop;
+    return String(prop);
```
Comment (JeanMeche) [inline devtools/projects/ng-devtools-backend/src/lib/state-serializer/serialized-descriptor-factory.ts:75]: nit: this should be a template literal, it's the most efficient way to create a string !

**Review by JeanMeche — APPROVED** [review]
Beside the small nit, LGTM !

File: devtools/projects/ng-devtools-backend/src/lib/state-serializer/serialized-descriptor-factory.ts (hunk 1 above)
Comment (ducin) [reply devtools/projects/ng-devtools-backend/src/lib/state-serializer/serialized-descriptor-factory.ts:75]: done

**Review by AleksanderBodurri — APPROVED** [review]
Looks great! Thank you @ducin 🙏

**Comment by ducin** [comment]
damn, missing semicolon and re-formatting. Fixed 🙃

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit cb500c7ee52497fe0e3aabbc52ac25b561e1d3f1.

**Pull request closed without merging**